### DISCUSSION TITLE
OTEL collector configuration with API key?

### DISCUSSION THREAD

**Discussion by Toudahl** [datalust/seq-tickets#2078] [Help]
Hello,

I am looking to utilize the otel collector for my hobby project.
I have it set up according to this: https://docs.datalust.co/docs/ingestion-with-opentelemetry#configuring-the-opentelemetry-collector

However, when i setup an api key and require that for ingestion, i get 403 in the collector.
I could add the key as a hardcoded value in the collector config. But then the collector would be application specific.

I had hoped that i could keep passing the key in the http header, and that it would be forwarded. But it seems that is not the case.

But perhaps i have misunderstood what it is that you mean when you say "application". So far i was thinking of api keys, as a per service thing. So each service would have their own.
But is the intention perhaps not as fine grained?

Im looking forward to your input, thanks :)

**Comment by KodrAus** [comment]
Hi @Toudahl :wave:

If one of the goals of your hobby project is specifically to work with the otel collector, or you have multiple destinations in play, then there does appear to be an extension you can use to forward headers through its pipeline: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/headerssetterextension. I haven’t used this extension myself so it might take a bit of trial-and-error to get working, but looks like it should do what you need.

If you just want to get diagnostics from your project into Seq, then pointing your OpenTelemetry configuration directly at Seq will likely be a lot less pain 🙂 And should be able to be configured to point at an otel collector at any point too.

**Reply by KodrAus** [reply]
> But perhaps i have misunderstood what it is that you mean when you say "application". So far i was thinking of api keys, as a per service thing. So each service would have their own.
But is the intention perhaps not as fine grained?

Also just to confirm this bit, you’re absolutely right there and the goal of API keys is to be pretty fine grained so you can enrich and throttle them independently 👍

**Reply by Toudahl** [reply]
I've had it working for a while with direct ingestion. But I'm planning on bringing my react client into my observability solution, and want to try to consolidate the processing of data in one place before it is forwarded to the various solutions I use for searching/visualisation.

Thanks for the link. I think it could work.
